src/compat/compat.h

```diff
@@ -3,21 +3,24 @@
 // Distributed under the MIT software license, see the accompanying
 // file COPYING or http://www.opensource.org/licenses/mit-license.php.
 
-#ifndef BITCOIN_COMPAT_H
-#define BITCOIN_COMPAT_H
+#ifndef BITCOIN_COMPAT_COMPAT_H
+#define BITCOIN_COMPAT_COMPAT_H
 
 #if defined(HAVE_CONFIG_H)
 #include <config/bitcoin-config.h>
 #endif
 
+// Windows defines FD_SETSIZE to 64, see _fd_types.h (mingw-w64),
+// which is too small for our usage. We redefine it to be 1024, to
+// match glibc, see typesizes.h.
```

Comment:
Some systems don't allow redefining FD_SETSIZE (notably including glibc), so IMO good to point out that Windows specifically allows it.

```suggestion
// Windows defines FD_SETSIZE to 64 (see _fd_types.h in mingw-w64),
// which is too small for our usage, but allows us to redefine it safely.
// We redefine it to be 1024, to match glibc, see typesizes.h.
```

Reply:
Taken the suggestion.